resolvers.ts

```diff
@@ -455,6 +460,211 @@ export const resolvers: Resolvers = {
         return res;
       }
     },
+    SampleForReport: async (root, args, context) => {
+      /* --------------------- Rails and Next Gen --------------------- */
+      const sampleInfo = await getFromRails({
+        url: `/samples/${args.railsSampleId}.json`,
+        args,
+        context,
+      });
+      if (sampleInfo?.pipeline_runs) {
+        const updatedPipelineRuns = sampleInfo?.pipeline_runs.map(
+          pipelineRun => {
+            return {
+              ...pipelineRun,
+              id: pipelineRun.id.toString(),
+            };
+          },
+        );
+        sampleInfo.pipeline_runs = updatedPipelineRuns;
+      }
+      if (sampleInfo?.workflow_runs) {
+        const updatedWorkflowRuns = sampleInfo?.workflow_runs.map(
+          workflowRun => {
+            return {
+              ...workflowRun,
+              id: workflowRun.id.toString(),
+            };
+          },
+        );
+        sampleInfo.workflow_runs = updatedWorkflowRuns;
+      }
+      if (sampleInfo?.project) {
+        sampleInfo.project.id = sampleInfo.project.id.toString();
+      }
+
+      const nextGenEnabled = await shouldReadFromNextGen(context);
+      /* --------------------- Rails --------------------- */
+      if (!nextGenEnabled) {
+        return {
+          id: `sample-for-query-${args.railsSampleId}`,
+          railsSampleId: args.railsSampleId,
+          ...sampleInfo,
+        };
+      }
+      /* --------------------- Next Gen --------------------- */
+
+      // NextGen Steps:
+      // continue using everything from rails except for workflow_runs
+      // get sample from rails using railsSampleId (same as above). This includes:
+      // -- sample info
+      // -- pipeline runs
+      // -- workflowRuns:
+      // ---- AMR workflow runs
+      // ---- pre-migration CG workflow runs
+      // ---- dual write CG workflow runs
+      // query entities using railsSampleId to get NextGenSampleId and done CG workflow runs
+      // query workflows using NextGenSampleId to get in progress CG workflow runs
+      // combine workflow data from entities and workflows
+      // deduplicate between rails and next gen
+
+      // TODO: 
+      // 1. find these in workflows now
+      // ✅ deprecated
+      // ✅ 🤷‍♀️ input_error
+      // 🤷‍♀️ run_finalized
+      // ✅ 🤷‍♀️ wdl_version
+
+      // 2. ⏰ parse ref_fasta from entitiesResp
+      // 3. ✅ 🤷‍♀️ specify "consensus-genome" workflow type from workflowsQuery
+      // 4. 🤷‍♀️ add clause to workflowQuery to specify "consensus-genome" rather than "bulk-download"
+      // 5. ✅ parse creationSource from rawInputsJson
+
+      const entitiesQuery = `
```

Comment:
This craziness makes sense to me. Sad that we have to do it but I don't see any other way 😢